homeassistant/components/formula_one/sensor.py

```diff
@@ -0,0 +1,419 @@
+"""Support for Formula 1 sensors."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import ergast_py as ergast
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, F1_DISCOVERY_NEW, F1_STATE_MULTIPLE
+from .coordinator import F1UpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Configure sensors from a config entry created in the integrations UI."""
+
+    @callback
+    def async_setup_sensors() -> None:
+        _LOGGER.debug("Handling new drivers/constructors/races")
+
+        coordinator: F1UpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+        entities: list[F1Sensor] = []
+
+        for standing in coordinator.constructor_standings:
+            new_cp_sensor = F1ConstructorPositionSensor(coordinator, standing.position)
+            if new_cp_sensor.unique_id not in coordinator.entity_ids:
+                entities.append(new_cp_sensor)
+
+            new_cn_sensor = F1ConstructorNameSensor(
+                coordinator,
+                standing.constructor.constructor_id,
+                standing.constructor.name,
+            )
+            if new_cn_sensor.unique_id not in coordinator.entity_ids:
+                entities.append(new_cn_sensor)
+
+        for standing in coordinator.driver_standings:
+            new_dp_sensor = F1DriverPositionSensor(coordinator, standing.position)
+            if new_dp_sensor.unique_id not in coordinator.entity_ids:
+                entities.append(new_dp_sensor)
+
+            new_dn_sensor = F1DriverNameSensor(
+                coordinator,
+                standing.driver.driver_id,
+            )
+            if new_dn_sensor.unique_id not in coordinator.entity_ids:
+                entities.append(new_dn_sensor)
+
+        for race in coordinator.races:
+            new_race_sensor = F1RaceSensor(coordinator, race.round_no)
+            if new_race_sensor.unique_id not in coordinator.entity_ids:
+                entities.append(new_race_sensor)
+
+        next_race_sensor = F1NextRaceSensor(coordinator)
+        if next_race_sensor.unique_id not in coordinator.entity_ids:
+            entities.append(next_race_sensor)
+
+        async_add_entities(entities)
+
+    async_setup_sensors()
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, F1_DISCOVERY_NEW, async_setup_sensors)
+    )
```

Comment:
As previously mentioned, do something like this instead
```suggestion
    coordinator.async_add_listener(async_setup_sensors)
```